### PR TITLE
[webview_flutter] Calling destroy on webview when flutter part is getting disposed

### PR DESCRIPTION
**Pull request by charafau**
## Description

Webview wasn't properly disposed on android

## Related Issues

fixes: https://github.com/flutter/flutter/issues/33299

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/webview_flutter/CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,11 @@
-## 0.3.10+3
+## 0.3.10+4
 
 * Don't log an unknown setting key error for 'debuggingEnabled' on iOS.
```
Comment (Liangdi) [inline packages/webview_flutter/CHANGELOG.md]: the wrong logs

Comment (charafau) [reply packages/webview_flutter/CHANGELOG.md]: thanks for noticing, I missed that after rebasing with master.

File: packages/webview_flutter/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
+## 0.3.10+4
+
+* Calling destroy on Android webview when flutter webview is getting disposed.
+
 ## 0.3.10+3
 
-* Don't log an unknown setting key error for 'debuggingEnabled' on iOS.
```
Comment (amirh) [inline packages/webview_flutter/CHANGELOG.md]: please revert

Comment (charafau) [reply packages/webview_flutter/CHANGELOG.md]: Sorry about confusion with changelog. I added `*` to that.

**Comment by charafau** [comment]
@amirh  I just noticed that you wanted to put some e2e test here. Not sure how I could test that. Any tips/ideas?

**Comment by charafau** [comment]
as far as I know there's no way of checking if sound still plays in webview.

**Comment by amirh** [comment]
> @amirh I just noticed that you wanted to put some e2e test here. Not sure how I could test that. Any tips/ideas?

You're right, I don't see an obvious way to test this, especially as we've lost the reference to the Android WebView after the platform view has been disposed.

Sorry about that.

**Comment by charafau** [comment]
Got it, thanks! let me rebase and let's merge.

**Comment by Brooooooklyn** [comment]
Why not the same logic in ios? I notice there are many of `com.apple.WebKit.WebContent` processes in my Mac `Activity monitor` while I was developing.

**Pull request merged**